### PR TITLE
Initial release

### PR DESCRIPTION
**Pull request by alexfedoseev**
@justin808 
This is loader for sass resources (e.g. variables, mixins etc.). Take a look.
P.S. I'll add basic README in a few hours.

### CODE REVIEW

**Comment by alexfedoseev** [comment]
[README.md](https://github.com/shakacode/sass-resources-loader/blob/alex/initial-release/README.md) added.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+// ...
+
+render() {
+  return (
+    <div className={css.section} />
+  );
+}
+```
+
+```scss
```
Comment (justin808) [inline README.md]: Maybe list this file before the JS file? and maybe use a different mixin name from the class name.

At first I read section thinking you were including the mixin as a class.

maybe use `section-mixin` for the mixin name.

File: scripts/build [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+$(npm bin)/babel \
```
Comment (justin808) [inline scripts/build]: Do you think it's better to keep the files separate or concatenate with webpack, as I do here: https://github.com/shakacode/node-webpack-async-example

File: scripts/clean [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+rm -rf lib
```
Comment (justin808) [inline scripts/clean]: why is the "dist" directory called "lib"? b/c that's how you want it to look in the published, babel converted one.

File: scripts/release [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+npm version $1 -m "Bump version to %s"
+git push
```
Comment (justin808) [inline scripts/release]: I think just the push with tags will push the other stuff? or maybe not?

Comment (justin808) [inline scripts/release]: btw - I use this: https://www.npmjs.com/package/release-it

File: src/loader.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  const sassResourcesIsString = typeof sassResources === 'string';
+  const sassResourcesIsArray = Array.isArray(sassResources);
+
+  logger.debug('sassResources is String:', sassResourcesIsString);
+  logger.debug('sassResources is Array:', sassResourcesIsArray);
+
+  if (!sassResourcesIsString && !sassResourcesIsArray) {
+    throw new Error(`
+      Looks like sassResources property has wrong type.
+      Make sure it's String or Array of Strings.
```
Comment (justin808) [inline src/loader.js]: do you want to check that we have an array of strings?

File: src/loader.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    `);
+  }
+
+  const result = [];
+
+  if (sassResourcesIsString) {
+    const sassResourcesImport = resolveResources(sassResources, this);
+    logger.debug('Resources:', sassResourcesImport || '**not found**');
```
Comment (justin808) [inline src/loader.js]: So logger, but not throw, just confirming

File: src/utils/fileExists.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import fs from 'fs';
+
+export default function(file) {
+  try {
+    return fs.statSync(file).isFile();
```
Comment (justin808) [inline src/utils/fileExists.js]: surprised there's no api for node that doesn't throw.

**Comment by justin808** [comment]
Looks great. I think we're about ready to ship!

File: README.md (hunk 1 above)
Comment (alexfedoseev) [inline README.md]: :+1:

File: scripts/build (hunk 2 above)
Comment (alexfedoseev) [inline scripts/build]: I use webpack for server-side code mainly for two reasons: better module isolation & CSS Modules. In this case babel is pretty straight, and I don't see profits of adding webpack layer. Is there any?

File: scripts/release (hunk 4 above)
Comment (alexfedoseev) [inline scripts/release]: It's nice one! :+1:

File: src/loader.js (hunk 5 above)
Comment (alexfedoseev) [inline src/loader.js]: Yep, missed it.

File: src/loader.js (hunk 6 above)
Comment (alexfedoseev) [inline src/loader.js]: I'm still not sure about this one, I think we should log it and throw (I'm not throwing here mainly b/c webpack will handle this throw in try/catch, and it's formatting is not cool for debug, but it noticeable).

**Pull request merged**